solr/core/src/java/org/apache/solr/cloud/api/collections/DeleteShardCmd.java

```diff
@@ -151,6 +154,21 @@ public void call(ClusterState clusterState, ZkNodeProps message, NamedList resul
           "Error executing delete operation for collection: " + collectionName + " shard: " + sliceId, e);
     }
   }
+  
+  private void cleanupZooKeeperShardMetadata(SolrZkClient client, String collection, String sliceId) throws InterruptedException {
+    String leaderElectPath = ZkStateReader.COLLECTIONS_ZKNODE + "/" + collection + "/leader_elect/" + sliceId;
+    String shardLeaderPath = ZkStateReader.COLLECTIONS_ZKNODE + "/" + collection + "/leaders/" + sliceId;
+    String shardTermsPath = ZkStateReader.COLLECTIONS_ZKNODE + "/" + collection + "/terms/" + sliceId;
+    
+    try {
+      client.clean(leaderElectPath);
+      client.clean(shardLeaderPath);
+      client.clean(shardTermsPath);
+    } catch (KeeperException ex) {
+      log.warn("Non-fatal error occured attempting to delete shard metadata on zooker for collection " + 
```

Comment:
If you are just logging a warning on failure, you might want to loop through each one, with the try-catch inside the loop. Therefore if one path fails, the others have a chance of succeeding. You can also log the path that failed which will help in debugging.